<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="meca_500_r3">

    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
    <group name="meca_arm">
        <!-- Defines the arm as a kinematic chain from its base to its mounting flange for the EEF -->
        <chain base_link="meca_base_link" tip_link="meca_axis_6_link" />
    </group>

    <group name="gripper">
        <!-- Defines the gripper group by its actuated joint(s) -->
        <link name="meca_gripper_link"/>
        <link name="meca_gripper_finger1"/>
        <link name="meca_gripper_finger2"/>
    </group>

    <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. -->
    <group_state name="zero" group="meca_arm">
        <!-- Only arm joints here -->
        <joint name="meca_axis_1_joint" value="0"/>
        <joint name="meca_axis_2_joint" value="0"/>
        <joint name="meca_axis_3_joint" value="0"/>
        <joint name="meca_axis_4_joint" value="0"/>
        <joint name="meca_axis_5_joint" value="0"/>
        <joint name="meca_axis_6_joint" value="0"/>
    </group_state>

    <group_state name="lowered" group="meca_arm">
        <!-- Only arm joints here -->
        <joint name="meca_axis_1_joint" value="0"/>
        <joint name="meca_axis_2_joint" value="1.3085"/>
        <joint name="meca_axis_3_joint" value="-0.4189"/>
        <joint name="meca_axis_4_joint" value="0"/>
        <joint name="meca_axis_5_joint" value="-0.9204"/>
        <joint name="meca_axis_6_joint" value="0"/>
    </group_state>

    <group_state name="center_beaker" group="meca_arm">
        <!-- Only arm joints here -->
        <joint name="meca_axis_1_joint" value="0"/>
        <joint name="meca_axis_2_joint" value="0.52"/>
        <joint name="meca_axis_3_joint" value="0.92"/>
        <joint name="meca_axis_4_joint" value="0"/>
        <joint name="meca_axis_5_joint" value="-1.448"/>
        <joint name="meca_axis_6_joint" value="0"/>
    </group_state>
    <group_state name="right_beaker" group="meca_arm">
        <!-- Only arm joints here -->
        <joint name="meca_axis_1_joint" value="0.314"/>
        <joint name="meca_axis_2_joint" value="0.52"/>
        <joint name="meca_axis_3_joint" value="0.91"/>
        <joint name="meca_axis_4_joint" value="0"/>
        <joint name="meca_axis_5_joint" value="-1.343"/>
        <joint name="meca_axis_6_joint" value="0"/>
    </group_state>

    <group_state name="gripper_closed" group="gripper"> <!-- Renamed for clarity -->
        <joint name="meca_gripper_finger1_joint" value="0"/>
    </group_state>

    <group_state name="gripper_open" group="gripper"> <!-- Renamed for clarity -->
        <joint name="meca_gripper_finger1_joint" value="0.024"/>
    </group_state>

    <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
    <end_effector name="tweezer_eef" parent_group="meca_arm" group="gripper" parent_link="meca_axis_6_link"/>

    <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
    <virtual_joint name="virtual_joint" type="fixed" parent_frame="world" child_link="meca_base_link"/>

    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
    <!-- Your original <disable_collisions> tags are likely fine, keeping them -->
    <disable_collisions link1="meca_axis_1_link" link2="meca_axis_2_link" reason="Adjacent"/>
    <disable_collisions link1="meca_axis_1_link" link2="meca_axis_3_link" reason="Never"/>
    <disable_collisions link1="meca_axis_1_link" link2="meca_base_link" reason="Adjacent"/>
    <disable_collisions link1="meca_axis_2_link" link2="meca_axis_3_link" reason="Adjacent"/>
    <disable_collisions link1="meca_axis_2_link" link2="meca_base_link" reason="Never"/>
    <disable_collisions link1="meca_axis_3_link" link2="meca_axis_4_link" reason="Adjacent"/>
    <disable_collisions link1="meca_axis_3_link" link2="meca_axis_5_link" reason="Never"/>
    <disable_collisions link1="meca_axis_3_link" link2="meca_axis_6_link" reason="Never"/>
    <disable_collisions link1="meca_axis_3_link" link2="meca_base_link" reason="Never"/>
    <disable_collisions link1="meca_axis_4_link" link2="meca_axis_5_link" reason="Adjacent"/>
    <disable_collisions link1="meca_axis_4_link" link2="meca_axis_6_link" reason="Never"/>
    <disable_collisions link1="meca_axis_4_link" link2="meca_gripper_finger1" reason="Never"/>
    <disable_collisions link1="meca_axis_4_link" link2="meca_gripper_finger2" reason="Never"/>
    <disable_collisions link1="meca_axis_5_link" link2="meca_axis_6_link" reason="Adjacent"/>
    <disable_collisions link1="meca_axis_5_link" link2="meca_gripper_finger1" reason="Never"/>
    <disable_collisions link1="meca_axis_5_link" link2="meca_gripper_finger2" reason="Never"/>
    <disable_collisions link1="meca_axis_5_link" link2="meca_gripper_link" reason="Default"/> <!-- Consider if this is "Never" or "Adjacent" -->
    <disable_collisions link1="meca_axis_6_link" link2="meca_gripper_finger1" reason="Never"/>
    <disable_collisions link1="meca_axis_6_link" link2="meca_gripper_finger2" reason="Never"/>
    <disable_collisions link1="meca_axis_6_link" link2="meca_gripper_link" reason="Adjacent"/> <!-- This is correct if gripper_link is rigidly attached to axis_6_link -->
    <disable_collisions link1="meca_gripper_finger1" link2="meca_gripper_finger2" reason="Never"/>
    <disable_collisions link1="meca_gripper_finger1" link2="meca_gripper_link" reason="Adjacent"/>
    <disable_collisions link1="meca_gripper_finger2" link2="meca_gripper_link" reason="Adjacent"/>
</robot>